oracle/datadog_checks/oracle/oracle.py

```diff
@@ -165,44 +141,37 @@ def check(self, _):
 
     @property
     def _connection(self):
-        """Creates a connection or raises an exception"""
         if self._cached_connection is None:
-            if self.can_use_oracle_client():
-                self._cached_connection = self._oracle_client_connect()
-            elif JDBC_IMPORT_ERROR:
-                self._connection_errors += 1
-                self.log.error(
-                    "Oracle client is unavailable and the integration is unable to import JDBC libraries. You may not "
-                    "have the Microsoft Visual C++ Runtime 2015 installed on your system. Please double check your "
-                    "installation and refer to the Datadog documentation for more information."
-                )
-                raise JDBC_IMPORT_ERROR
+            if self.use_jdbc():
+                try:
+                    self._cached_connection = self._jdbc_connect()
+                except Exception:
+                    self.log.error(
+                        "The integration is unable to import JDBC libraries. You might not "
+                        "have the Microsoft Visual C++ Runtime 2015 installed on your system. Please double check your "
+                        "installation and refer to the Datadog documentation for more information."
+                    )
+                    raise JDBC_IMPORT_ERROR
             else:
-                self._cached_connection = self._jdbc_connect()
+                self._cached_connection = self._oracle_connect()
         return self._cached_connection
 
-    def can_use_oracle_client(self):
-        try:
-            # Check if the instantclient is available
-            cx_Oracle.clientversion()
-        except cx_Oracle.DatabaseError as e:
-            # Fallback to JDBC
-            self.log.debug('Oracle instant client unavailable, falling back to JDBC: %s', e)
-            return False
-        else:
-            self.log.debug('Running cx_Oracle version %s', cx_Oracle.version)
+    def use_jdbc(self):
+        if self._jdbc_driver and not JDBC_IMPORT_ERROR:
```

Comment:
if `jdbc_driver_path` is set and there is a JDBC_IMPORT_ERROR we should raise an error